### PR TITLE
New Pattern: AWS Lambda REST API behind Amazon ALB with resource-path Listener Rule (Python & SAM)

### PR DESCRIPTION
**Pull request by bb245**
Issue https://github.com/aws-samples/serverless-patterns/issues/1991

Description of changes:
This setup orchestrates the deployment of an Application Load Balancer, configures path-based routes directing traffic to a Python-based AWS Lambda function, and leverages the [AWS Lambda Powertools for Python](https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/) library. The Lambda function, serving as the target, records details of the incoming ALB event, along with the API and context objects, logging them to an Amazon CloudWatch Logs log group and Amazon X-Ray.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitignore
Comment (bfreiberg) [inline .gitignore:1]: Please revert the changes to this global file

File: alb-lambda-rest-api-sam-py/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# AWS Lambda RESTful API with Amazon ALB and Path Based Listener Rules
+
+This configuration employs Python 3.9 and the Serverless Application Model (SAM) CLI to establish an Application Load
+Balancer with path-based listener rules, paired with an AWS Lambda RESTful API function as the target.
```
Comment (bfreiberg) [inline alb-lambda-rest-api-sam-py/README.md:4]:
```suggestion
This configuration pattern creates an Application Load
Balancer with path-based listener rules, paired with an AWS Lambda RESTful API function as the target. It uses  Python 3.9 and the Serverless Application Model (SAM) CLI.
```

File: alb-lambda-rest-api-sam-py/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/alb-lambda-rest-api-sam-py",
+      "templateURL": "serverless-patterns/alb-lambda-rest-api-sam-py",
+      "projectFolder": "alb-lambda-rest-api-sam-py",
+      "templateFile": "template.yaml"
+    }
+  },
+  "resources": {
```
Comment (bfreiberg) [inline alb-lambda-rest-api-sam-py/example-pattern.json:21]: please include a link to Lambda Powertools (even though there is already a link in the introBox)

File: alb-lambda-rest-api-sam-py/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    ```
+3. Install dependencies and build:
+    ```
+    pipenv install
+    ```
+4. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam validate && sam build && sam deploy --guided
+    ```
+5. During the prompts:
```
Comment (bfreiberg) [inline alb-lambda-rest-api-sam-py/README.md:43]: When I run `sam deploy --guided` the  stack also asks me for a VPC id and subnets. Please include instructions what the user needs to do here and what format is expected.

**Comment by bb245** [comment]
@bfreiberg thanks for the review. I've updated the codebase accordingly.

File: alb-lambda-rest-api-sam-py/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```
+    cd alb-lambda-rest-api-sam-py
+    ```
+3. Install dependencies and build:
+    ```
+    pipenv install
```
Comment (bfreiberg) [inline alb-lambda-rest-api-sam-py/README.md:38]: If I do not run `pipenv requirements > requirements.txt` before `sam deploy` the Powertools dependency is not installed. Is the command missing here or did I miss something?

File: alb-lambda-rest-api-sam-py/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
+{
+  "title": "AWS Lambda REST API and Amazon ALB with Listener Rules",
+  "description": "This pattern creates an Application Load Balancer with route-based listener rules, paired with an AWS Lambda RESTful API function as the target. It uses Python 3.9 and the Serverless Application Model (SAM) CLI.",
```
Comment (bfreiberg) [inline alb-lambda-rest-api-sam-py/example-pattern.json:3]: The description has a maximum length of 150 characters, please shorten it a bit, for example

> This pattern creates an Application Load Balancer with route-based listener rules, paired with an AWS Lambda function as the target.

as the usage of SAM and Python is expressed via the `language` and `framework` fields already

File: alb-lambda-rest-api-sam-py/README.md (hunk 4 above)
Comment (bb245) [reply alb-lambda-rest-api-sam-py/README.md:38]: I had this step in Testing the example locally. But it makes sense to have the instruction here as well.

**Comment by bb245** [comment]
@bfreiberg, fyi please, i've addressed a lint issue and tested the change.

**Comment by bfreiberg** [comment]
Looks good, thank you for your contribution.

**Comment by bb245** [comment]
> Looks good, thank you for your contribution.

Thank you!
